src/instructions.rs

```diff
@@ -0,0 +1,814 @@
+    Ok(())
+}
+
+/// Loads a value that is located in a memory address, formed by the addition
+/// of the value on a register and in the offset6 section, into a desired register
+pub fn load_register(instr: u16, regs: &mut Registers, memory: &mut Memory) -> Result<(), VMError> {
+    // Destination Register
+    let dr = Register::from_u16((instr >> 9) & THREE_BIT_MASK)?;
+    // BaseR section
+    let r1 = Register::from_u16((instr >> 6) & THREE_BIT_MASK)?;
+    // Offset6 section
+    let mut offset6 = instr & SIX_BIT_MASK;
+    offset6 = sign_extend(offset6, 6)?;
+    // Calculate the memory address to read
+    let address = regs[r1].wrapping_add(offset6);
+    regs[dr] = memory.read(address)?;
+    update_flags(dr, regs);
+    Ok(())
+}
+
+/// Loads a value created from the addition of the value of the PC and the
+/// one in the PCoffset9 section, into a register
+pub fn load_effective_address(instr: u16, regs: &mut Registers) -> Result<(), VMError> {
+    // Destination Register
+    let dr = Register::from_u16((instr >> 9) & THREE_BIT_MASK)?;
+    // PCoffset9 section
+    let mut pc_offset = instr & NINE_BIT_MASK;
+    pc_offset = sign_extend(pc_offset, 9)?;
+    // Set the new value for the destination register
+    regs[dr] = regs[Register::PC].wrapping_add(pc_offset);
+    update_flags(dr, regs);
+    Ok(())
+}
+
+/// Stores the value that is in a register into an address in memory. This address
+/// is created from the addition of the PC and the PCoffset9 section
+pub fn store(instr: u16, regs: &mut Registers, memory: &mut Memory) -> Result<(), VMError> {
+    // Source Register
+    let sr = Register::from_u16((instr >> 9) & THREE_BIT_MASK)?;
+    // PCoffset9 section
+    let mut pc_offset = instr & NINE_BIT_MASK;
+    pc_offset = sign_extend(pc_offset, 9)?;
+    // Calculate the address
+    let address = regs[Register::PC].wrapping_add(pc_offset);
+    let new_val = regs[sr];
+    memory.write(address, new_val)
+}
+
+/// Stores the value that is in a register into an address in memory. This address
+/// is taken indirectly from the instruction. By adding the PC and the PCoffset9 section
+/// we get the first memory address, then if we read it we get the final address. That
+/// final address is the one that is going to get written.
+pub fn store_indirect(
+    instr: u16,
+    regs: &mut Registers,
+    memory: &mut Memory,
+) -> Result<(), VMError> {
+    // Source Register
+    let sr = Register::from_u16((instr >> 9) & THREE_BIT_MASK)?;
+    // PCoffset9 section
+    let mut pc_offset = instr & NINE_BIT_MASK;
+    pc_offset = sign_extend(pc_offset, 9)?;
+    // Get the first address
+    let first_address = regs[Register::PC].wrapping_add(pc_offset);
+    // Read the first address, get the second one and write on it
+    let final_address = memory.read(first_address)?;
+    let new_val = regs[sr];
+    memory.write(final_address, new_val)
+}
+
+/// Stores the value that is in a register into an address in memory. By adding
```

Comment:
The value is not stored in an address but in memory:

```suggestion
/// Reads a value from a register and stores it into memory. This address
```

Then you can describe where the value is stored.